### PR TITLE
Add optional scaling to RRTMGP flux adjustment

### PR DESCRIPTION
**Pull request by dustinswales**
## Description
This PR contains improvements to how the LW heating-rate profile is updated between calls to the RRTMGP radiation scheme.
The adjusted net LW flux is computed following Hogan and Bozzo 2015 (10.1002/2015MS000455). Optionally, the flux adjustment can be damped with height using a logistic function. 

## Testing
This was tested on Hera using the Intel compilers. Baselines for the c192l127 RRTMGP regression test will need to be updated. /scratch2/BMC/ome/Dustin.Swales/UFS/stmp2/Dustin.Swales/FV3_RT/rt_279869/fv3_gfs_v16_RRTMGP_c192L127

## Dependencies
https://github.com/dustinswales/ccpp-physics/tree/hotfix_AddPLimitToLWJacAdj
## Waiting for other repos
https://github.com/NCAR/ccpp-physics/pull/668

### CODE REVIEW

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -3790,6 +3798,9 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%doGP_cldoptics_PADE = doGP_cldoptics_PADE
     Model%doGP_cldoptics_LUT  = doGP_cldoptics_LUT
     Model%use_LW_jacobian     = use_LW_jacobian
+    Model%damp_LW_fluxadj     = damp_LW_fluxadj
+    Model%lfnc_k              = 1. / lfnc_k
```
Comment (climbfuji) [inline ccpp/data/GFS_typedefs.F90]: Now this is pretty confusing ... why can't `lfnc_k` be defined such that `Model%lfnc_k = lfnc_k`?  And would a default value of `1.0/-999.0=-0.001...` still be correct (I assume unused if `< 0`?)

**Comment by yangfanglin** [comment]
Dustin,

I'd like to confirm, per our email communications, the hard-wired cutoff adjustment at 100hPa will not be used.  The default will the flux adjustment  damped with height using a logistic function (the green line from your plots).

File: ccpp/data/GFS_typedefs.F90 (hunk 1 above)
Comment (dustinswales) [reply ccpp/data/GFS_typedefs.F90]: You're spot on. I just cleaned this up

**Review by climbfuji — APPROVED** [review]

**Comment by dustinswales** [comment]
> Dustin,
> 
> I'd like to confirm, per our email communications, the hard-wired cutoff adjustment at 100hPa will not be used. The default will the flux adjustment damped with height using a logistic function (the green line from your plots).

This is correct.
The logistic function is controlled by the namelist, https://github.com/dustinswales/ufs-weather-model/blob/hotfix_AddPLimitToLWJacAdj/tests/parm/v16_c192L127_rrtmgp.nml.IN#L208. The adjustment damping is centered at 600mb. The k value is in inverse pressure distance, 1/31.25mb. 

Some additional info.
I tested a few different configurations with c192l127
flxadj_hb2015_lower
On the left is the mean change (all profiles from 2 day forecast) in the LW heating-rate due to the flux adjustment. The thick black line is without any logistic function damping applied.
On the right are several logistic functions, then overlaid on the left panel (sorry for the poor plot quality) are the impacts of the damping. 
Then on the full profile,
flxadj_hb2015

**Comment by yangfanglin** [comment]
Thanks Dustin.  These plots are very useful for understanding the impact.  The model should be more stable with the new approach.

**Comment by climbfuji** [comment]
This PR has been pulled into https://github.com/NOAA-EMC/fv3atm/pull/331 and will be merged as part of it.

**Comment by climbfuji** [comment]
https://github.com/NOAA-EMC/fv3atm/pull/331 was merged and included all changes in this PR

**Pull request closed without merging**